backend/fms_core/migrations/0066_v4_12_1.py

```diff
@@ -0,0 +1,95 @@
+                                       needs_placement=False,
+                                       needs_planning=False,
+                                       expected_sample_type=step_info["expected_sample_type"],
+                                       created_by_id=admin_user.id,
+                                       updated_by_id=admin_user.id)
+            reversion.add_to_revision(step)
+            for specification in step_info["specifications"]:
+                step_specification = StepSpecification.objects.create(name=specification["display_name"],
+                                                                      sheet_name=specification["sheet_name"],
+                                                                      column_name=specification["column_name"],
+                                                                      value=specification["value"],
+                                                                      step=step,
+                                                                      created_by_id=admin_user.id,
+                                                                      updated_by_id=admin_user.id)
+                reversion.add_to_revision(step_specification)
+        oldStep = Step.objects.get(name="Sample QC")
+        dnaStep = Step.objects.get(name="Sample QC (RNA)")
+        rnaStep = Step.objects.get(name="Sample QC (DNA)")
+        # Start for data transfer process
+        sns = SampleNextStep.objects.filter(step__id=oldStep.id)
+        for sampleNextStep in sns:
+            updatedSampleNextStep = SampleNextStep.objects.get(id=sampleNextStep.id)
+            sampleNextStepByStudy = SampleNextStepByStudy.objects.get(sample_next_step__id=sampleNextStep.id)
+            stepOrder = StepOrder.objects.get(id=sampleNextStepByStudy.step_order.id)
+            try:
+              if sampleNextStep.sample.derived_samples.first().sample_kind.name == "DNA":
+                updatedSampleNextStep.step = dnaStep
+                stepOrder.step = dnaStep
+              elif sampleNextStep.sample.derived_samples.first().sample_kind.name == "RNA":
+                updatedSampleNextStep.step = rnaStep
+                stepOrder.step = rnaStep
+            except Exception:
+              errors["AssigningError"] = f"There was an error while assigning the new steps to the existing sample next steps objects {sampleNextStep}."
+            if errors:
+                raise ValidationError(errors)
+            stepOrder.save()
+            updatedSampleNextStep.save()
+            reversion.add_to_revision(updatedSampleNextStep)
+            reversion.add_to_revision(stepOrder)
+        # make sure there are no loose ends before deleting oldStep
+        so = StepOrder.objects.filter(step__id=oldStep.id)
+        for order in so:
+            stepOrder = StepOrder.objects.get(id=order.id)
+            if "DNA" in order.previous_step_order.all().first().step.name:
```

Comment:
this is risky. 3 letters could be part of normal words (BANDNAME !)... this likely work but it would be much safer to use the full step name.